src/DateTimeParser.jl

```diff
@@ -0,0 +1,512 @@
+module DateTimeParser
+
+using Base.Dates
+using TimeZones
+import Base.Dates: VALUETODAYOFWEEK, VALUETODAYOFWEEKABBR, VALUETOMONTH, VALUETOMONTHABBR
+export parsedate
```

Comment:
I think this module should export the function `parse` which will have these methods defined:
```
parse(::Type{ZonedDateTime}, str::AbstractString)
parse(::Type{DateTime}, str::AbstractString)
parse(::Type{Date}, str::AbstractString)
```